File: app/src/main/java/com/sevtinge/hyperceiler/module/hook/mms/DisableAd.java
/*
  * This file is part of HyperCeiler.

  * HyperCeiler is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Affero General Public License as
  * published by the Free Software Foundation, either version 3 of the
  * License.

  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Affero General Public License for more details.

  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.

  * Copyright (C) 2023-2024 HyperCeiler Contributions
*/
package com.sevtinge.hyperceiler.module.hook.mms;

import android.content.Context;

import com.sevtinge.hyperceiler.module.base.BaseHook;
import com.sevtinge.hyperceiler.module.base.dexkit.DexKit;
import com.sevtinge.hyperceiler.module.base.dexkit.IDexKit;
import com.sevtinge.hyperceiler.module.base.dexkit.IDexKitList;

import org.luckypray.dexkit.DexKitBridge;
import org.luckypray.dexkit.query.FindMethod;
import org.luckypray.dexkit.query.matchers.ClassMatcher;
import org.luckypray.dexkit.query.matchers.MethodMatcher;
import org.luckypray.dexkit.result.MethodData;
import org.luckypray.dexkit.result.MethodDataList;

import java.lang.reflect.AnnotatedElement;
import java.lang.reflect.Method;
import java.lang.reflect.Modifier;
import java.util.List;

public class DisableAd extends BaseHook {
    @Override
    public void init() throws NoSuchMethodException {

        findAndHookMethod("com.miui.smsextra.ui.BottomMenu", "allowMenuMode",
            Context.class, new MethodHook() {
                @Override
                protected void before(MethodHookParam param) throws Throwable {
                    param.setResult(false);
                }
            });

        List<Method> methods = DexKit.getDexKitBridgeList("HideButton", new IDexKitList() {
            @Override
            public List<AnnotatedElement> dexkit(DexKitBridge bridge) throws ReflectiveOperationException {
                MethodDataList methodData = bridge.findMethod(FindMethod.create()
                        .matcher(MethodMatcher.create()
                                .name("setHideButton")
                        )
                );
                return DexKit.toElementList(methodData, lpparam.classLoader);
            }
        }).toMethodList();
        for (Method method2 : methods) {
                    if (!Modifier.isAbstract(method2.getModifiers())) {
                        hookMethod(method2, new MethodHook() {
                            @Override
                            protected void before(MethodHookParam param) throws Throwable {
                                param.args[0] = true;
                            }
                        });
                    }
        }
}
}

File: app/src/main/java/com/sevtinge/hyperceiler/module/hook/mms/DisableAd.kt
package com.sevtinge.hyperceiler.module.hook.mms

import android.content.Context
import com.github.kyuubiran.ezxhelper.EzXHelper
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHooks
import com.github.kyuubiran.ezxhelper.finders.MethodFinder.`-Static`.methodFinder
import com.sevtinge.hyperceiler.module.base.BaseHook
import com.sevtinge.hyperceiler.utils.DexKit.addUsingStringsEquals
import com.sevtinge.hyperceiler.utils.DexKit.dexKitBridge
import java.lang.reflect.Modifier


object DisableAd : BaseHook() {
    override fun init() {
        try {
            dexKitBridge.findMethod {
                matcher {
                    declaredClass {
                        addUsingStringsEquals("Unknown type of the message: ")
                    }
                    usingNumbers(3, 4)
                    modifiers = Modifier.FINAL
                    returnType = "boolean"
                    paramCount = 0
                }
            }.map { it.getMethodInstance(EzXHelper.safeClassLoader) }.createHooks {
                returnConstant(false)
            }
        } catch (e: Throwable) {
            logE(TAG, this.lpparam.packageName, e)
        }
        findAndHookMethod("com.miui.smsextra.ui.BottomMenu", "allowMenuMode",
            Context::class.java, object : MethodHook() {
                @Throws(Throwable::class)
                override fun before(param: MethodHookParam) {
                    param.setResult(false)
                }
            })
    }
}
